action/action.yml

```
@@ -28,14 +28,19 @@ runs:
         VERSION=${{ inputs.version }}
         TOKEN=${{ inputs.token }}
 
-        if [ -n "${TOKEN}" ]; then
-          TOKEN=(-H "Authorization: token ${TOKEN}")
-        fi
-
         if [ -z "${VERSION}" ]; then
-          VERSION=$(curl https://api.github.com/repos/fluxcd/flux2/releases/latest -sL ${TOKEN[@]} | grep tag_name | sed -E 's/.*"([^"]+)".*/\1/' | cut -c 2-)
+          if [ -n "${TOKEN}" ]; then
+            TOKEN=(-H "Authorization: token ${TOKEN}")
+            VERSION_SLUG=$(curl https://api.github.com/repos/fluxcd/flux2/releases/latest -sL "${TOKEN[@]}" | grep tag_name)
```

Are you comfortable eliminating this intermediate array?
```suggestion
            VERSION_SLUG=$(curl https://api.github.com/repos/fluxcd/flux2/releases/latest --silent --location --header "Authorization: token ${TOKEN}" | grep tag_name)
```
Also, consider hoisting out the common _grep tag_name_ invocation from lines 34 and 38, moving it down into the pipeline at line 42.

---

I'd find this clearer to read if the URL was the last command-line argument passed to _curl_, with the other flags preceding it.

---

Fixed in 86e9d40